### ISSUE TITLE
clang 14 not support C++17 fully

### ISSUE THREAD

**Issue by sidyhe** [shepherdjerred/macos-cross-compiler#2]
my code use <memory_resource> but it still in experimental.
i tested on macos 11 with clang 19 (by ``brew install llvm``), it is ok
the cppreference say it support until clang15 
so if clang in docker can be up to 15, is better for C++17

https://en.cppreference.com/w/cpp/compiler_support/17

**Comment by shepherdjerred** [comment]
Hi! Can you provide a small program (ideally a single file) so I can test with?

**Comment by shepherdjerred** [comment]
I've updated the image to use gcc 14, clang 19, and macOS 15.1 in https://github.com/shepherdjerred/macos-cross-compiler/commit/7719cdb68d490767c6cedc977b0e1af146d95e52

You can try the latest Docker image once it's built.

**Comment by sidyhe** [comment]
thank you, i am pulling this image, but slowly

i was modified code for this case, build ok for x64 amd arm64
and i see the latest tag is remove support for x64 ....
my code mainly for win32 and linux (x86/x64), some days ago i want to build for macos 11 x64 and arm64 for widely running

in summary, i wish your [macos-cross-compiler](https://github.com/shepherdjerred/macos-cross-compiler) can build for macos 11 x64 and arm64 with full C++17 support

**Comment by sidyhe** [comment]
```
        std::array<int, 32> mem;
        std::pmr::monotonic_buffer_resource pool(mem.data(), mem.size() * sizeof(mem[0]));
        std::pmr::vector<int> drlst(&pool);
```

**Comment by sidyhe** [comment]
ok! thank you!
the image (tag 15.0) with clang 18.1.3 is good that implement <memory_resource>
build for arm64 success, but x64 not support...

by the way, no git and cmake in it, so i need ``apt install git cmake`` every time

**Comment by shepherdjerred** [comment]
I can look into re-adding x86_64 support. I felt it wasn't necessary since Apple is well into x86_64 deprecation. 

> by the way, no git and cmake in it, so i need apt install git cmake every time

I'd recommend creating your own Dockerfile using this as the base image so that you can add whatever you might need.

**Comment by sidyhe** [comment]
thank you

**Comment by shepherdjerred** [comment]
Hi! I was able to get this working without too much effort. [This](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=116809) was the root issue which I resolved by using @iains's build of GCC instead of GCC from Ubuntu Noble. I was already using their build of GCC for arm64, so now I'm just using it for both arm64 and x86_64.

Anyway, if you pull the latest image it should just work.

> and every command has a warning
> osxcross: warning: '-mmacosx-version-min=' (11.0.0 != 11)

I'll look into this later, but it seems harmless.
